### PR TITLE
NetworkHelper: Use it where possible

### PR DESCRIPTION
**Pull request by REASY**
- `NetworkHelper` use only array under the hood
- Move often call to get link to `NetworkHelper`

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1217)

### CODE REVIEW

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Test!

**Comment by colinsheppard** [comment]
test!

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
Test!

File: src/main/scala/beam/utils/ProfilingUtils.scala [hunk 1]
```diff
@@ -15,4 +15,10 @@ object ProfilingUtils {
     val endTime = System.currentTimeMillis()
     endTime - startTime
   }
+
+  def timed[U](what: String, work: => U, logger: String => Unit): U = {
```
Comment (JustinPihony) [inline src/main/scala/beam/utils/ProfilingUtils.scala]: What about changing the method signature to:

    def timed[U](what: String, logger: String => Unit)(work: () => U): U = {

As you can then use it like this:

    time("what", x=>log.info(x)){ () => 
      //inline method
    }

Not a huge difference, so take it or leave it. I just know that this is often the pattern for this type of method.

**Comment by REASY** [comment]
I was confused by the logs, it says:
```
13:42:58 06:42:58.571 [PersonAgentSpec-akka.actor.default-dispatcher-10] ERROR akka.actor.OneForOneStrategy - null
13:42:58 java.lang.NullPointerException: null
13:42:58 	at beam.agentsim.agents.vehicles.BeamVehicle$.$anonfun$collectFuelConsumptionData$2(BeamVehicle.scala:221)
13:42:58 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
13:42:58 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:58)
```
** PersonAgentSpec-akka.actor.default-dispatcher-10**. When I run `PersonAgentSpec` test on local machine `./gradlew specificTest -Psuite="beam.agentsim.agents.PersonAgentSpec"`, it passes all the test. When I looked to the code, I found that `PersonAgentSpec` name for the actor system was used also for `PersonAndTransitDriverSpec`, `PersonWithCarPlanSpec` and `PersonWithVehicleSharingSpec`. Has to fix the mock there.

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Test!

**Comment by colinsheppard** [comment]
test!

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
I can see exception during handling events (https://beam-ci.tk/blue/rest/organizations/jenkins/pipelines/beam-4ci/runs/4227/log/?start=0):
```
15:28:30.669 [Events-0] ERROR o.m.c.e.ParallelEventsManagerImpl - Thread Events-0 died with exception while handling events.
java.lang.ArrayIndexOutOfBoundsException: 379
	at beam.analysis.DelayMetricAnalysis.process(DelayMetricAnalysis.scala:103)
	at beam.analysis.DelayMetricAnalysis.handleEvent(DelayMetricAnalysis.scala:74)
	at org.matsim.core.events.EventsManagerImpl.callHandlerFast(EventsManagerImpl.java:333)
	at org.matsim.core.events.EventsManagerImpl.computeEvent(EventsManagerImpl.java:222)
	at org.matsim.core.events.EventsManagerImpl.processEvent(EventsManagerImpl.java:136)
	at org.matsim.core.events.ProcessEventThread.run(ProcessEventThread.java:93)
	at java.lang.Thread.run(Thread.java:748)
```
Need to run it in debug mode to see what went wrong.

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Test !

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Previous build was killed because run is stuck (https://beam-ci.tk/blue/rest/organizations/jenkins/pipelines/beam-4ci/runs/4272/log/?start=0) and stuck finder is turned-off for beam run (https://github.com/LBNL-UCB-STI/beam/blob/art/use-network-helper-4ci/test/input/beamville/beam.conf#L176).
Going to turn it on and wait for build result

**Comment by REASY** [comment]
Test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Pull request merged**